core/src/main/java/org/infinispan/interceptors/locking/OptimisticLockingInterceptor.java

```diff
@@ -139,6 +89,28 @@ public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) t
    }
 
    @Override
+   public Object visitLockControlCommand(TxInvocationContext ctx, LockControlCommand command) throws Throwable {
+      throw new InvalidCacheUsageException("Explicit locking is not allowed with optimistic caches!");
+   }
+
+   @Override
+   protected Log getLog() {
+      return log;
+   }
```

Comment:
I only noticed that it was moved, I didn't realize the other visit methods aren't from the `Visitor` interface :)